cli/cmd/verify/component-version/cmd.go

```diff
@@ -0,0 +1,318 @@
+	}
+
+	verifierSpecPath, err := cmd.Flags().GetString(FlagVerifierSpec)
+	if err != nil {
+		return fmt.Errorf("getting verifier-spec flag failed: %w", err)
+	}
+
+	reference := args[0]
+	config := ocmContext.Configuration()
+
+	//nolint:staticcheck // no replacement for resolvers available yet
+	var resolvers []*resolverruntime.Resolver
+	if config != nil {
+		resolvers, err = ocm.ResolversFromConfig(config)
+		if err != nil {
+			return fmt.Errorf("getting resolvers from configuration failed: %w", err)
+		}
+	}
+	repo, err := ocm.NewFromRefWithFallbackRepo(ctx, pluginManager, credentialGraph, resolvers, reference)
+	if err != nil {
+		return fmt.Errorf("could not initialize ocm repository: %w", err)
+	}
+
+	desc, err := repo.GetComponentVersion(ctx)
+	if err != nil {
+		return fmt.Errorf("getting component reference and versions failed: %w", err)
+	}
+
+	var sigs []descruntime.Signature
+	if signatureName != "" {
+		for _, sig := range desc.Signatures {
+			if sig.Name == signatureName {
+				sigs = append(sigs, sig)
+				break
+			}
+		}
+	} else {
+		sigs = desc.Signatures
+	}
+
+	if len(sigs) == 0 {
+		return fmt.Errorf("no signatures found to verify")
+	}
+
+	if verifyDigestConsistency {
+		if err := signing.IsSafelyDigestible(&desc.Component); err != nil {
+			logger.WarnContext(ctx, "component version is not considered safely digestable", "error", err.Error())
+		}
+		for _, resource := range desc.Component.Resources {
+			if err := verifyResourceDigest(cmd, resource, pluginManager, credentialGraph); err != nil {
+				return fmt.Errorf("resource digest verification failed: %w", err)
+			}
+		}
+	}
+
+	var verifierSpec runtime.Typed
+	if verifierSpecPath == "" {
+		logger.InfoContext(ctx, "no verifier specification file given, using default RSASSA-PSS")
+		verifierSpec = &v1alpha1.Config{}
+		_, _ = v1alpha1.Scheme.DefaultType(verifierSpec)
+	} else {
+		genericScheme := runtime.NewScheme(runtime.WithAllowUnknown())
+		verifierSpecBytes, err := os.ReadFile(verifierSpecPath)
+		if err != nil {
+			return fmt.Errorf("reading verifier specification file %q failed: %w", verifierSpecPath, err)
+		}
+		verifierSpec = &runtime.Raw{}
+		if err := genericScheme.Decode(bytes.NewReader(verifierSpecBytes), verifierSpec); err != nil {
+			return fmt.Errorf("decoding verifier specification file %q failed: %w", verifierSpecPath, err)
+		}
+	}
+
+	handler, err := pluginManager.SigningRegistry.GetPlugin(ctx, verifierSpec)
+	if err != nil {
+		return fmt.Errorf("getting signature handler plugin failed: %w", err)
+	}
+
+	eg, egctx := errgroup.WithContext(ctx)
+	eg.SetLimit(concurrencyLimit)
+	for _, signature := range sigs {
+		s := signature
+		eg.Go(func() error {
+			start := time.Now()
+			logger.InfoContext(egctx, "verifying signature", "name", s.Name)
+			defer func() {
+				logger.InfoContext(egctx, "signature verification completed", "name", s.Name, "duration", time.Since(start).String())
+			}()
```

Comment:
I think since go 1.22, loop variables have per iteration scope (https://tip.golang.org/doc/go1.22#language / https://go.dev/blog/loopvar-preview), so this could theoretically be omitted.

```suggestion
	for _, signature := range sigs {
		eg.Go(func() error {
			start := time.Now()
			logger.InfoContext(egctx, "verifying signature", "name", signature.Name)
			defer func() {
				logger.InfoContext(egctx, "signature verification completed", "name", signature.Name, "duration", time.Since(start).String())
			}()
```